agreementbot/governance.go

```diff
@@ -210,7 +211,14 @@ func (w *AgreementBotWorker) GovernAgreements() int {
 			}
 
 		} else {
-			glog.Errorf(logString(fmt.Sprintf("unable to read agreements from database, error: %v", err)))
+			msg := logString(fmt.Sprintf("unable to read agreements from database, error: %v", err))
+			glog.Errorf(msg)
+
+			// This is the case where Postgresql database certificate got upgraded. The error is: "x509: certificate signed by unknown authority"
+			// Only checks for "x509" here to support globalization.
+			if strings.Contains(err.Error(), "x509") || strings.Contains(err.Error(), "X509") {
```

Comment:
The logs show there were several other places that could detect the x509 error. Is there a reason to only check this one place in governance and not check the others?  Possibly add a log message here too saying you will panic due to a certificate error.

Reply:
Only checking this place because this function (GovernAgreements()) is called periodically.

Reply:
I will change the code to exit out instead of panic.

Reply:
the partition heartbeat check is also called periodically... Perhaps those 2 places would be sufficient since they are called at regular intervals